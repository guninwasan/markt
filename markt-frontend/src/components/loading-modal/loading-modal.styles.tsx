import styled from "styled-components";
import { colors } from "../../utils";

const ModalBackground = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.7);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1000;
  flex-direction: column;
  gap: 1rem;
`;

const LoadingIndicator = styled.div`
  width: 100px;
  height: 100px;
  border: 7px solid ${colors.lightGrey};
  border-top: 10px solid ${colors.darkerHoverPrimary};
  border-radius: 50%;
  animation: spin 1s linear infinite;

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

const LoadingText = styled.div`
  width: 100%;
  font-weight: bold;
  color: ${colors.white};
  font-size: 1.5rem;
  font-weight: bold;
  text-align: center;
  justify-content: center;
  align-items: center;
  align-self: center;
  margin-left: 1.4rem;
`;

export { ModalBackground, LoadingIndicator, LoadingText };
