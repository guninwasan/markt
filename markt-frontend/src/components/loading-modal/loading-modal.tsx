import React from "react";
import {
  ModalBackground,
  LoadingIndicator,
  LoadingText,
} from "./loading-modal.styles";

const LoadingModal = () => {
  return (
    <ModalBackground data-testid="modal-background">
      <LoadingIndicator data-testid="loading-indicator" />
      <LoadingText>Loading....</LoadingText>
    </ModalBackground>
  );
};

export { LoadingModal };
