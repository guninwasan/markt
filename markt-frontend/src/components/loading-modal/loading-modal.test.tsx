import React from "react";
import { render, screen } from "@testing-library/react";
import { LoadingModal } from "./loading-modal";

describe("LoadingModal", () => {
  it("should render the modal background", () => {
    render(<LoadingModal />);
    expect(screen.getByTestId("modal-background")).toBeInTheDocument();
  });

  it("should render the loading indicator", () => {
    render(<LoadingModal />);
    expect(screen.getByTestId("loading-indicator")).toBeInTheDocument();
  });

  it("should render the loading text", () => {
    render(<LoadingModal />);
    expect(screen.getByText("Loading....")).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const { container } = render(<LoadingModal />);
    expect(container).toMatchSnapshot();
  });
});
