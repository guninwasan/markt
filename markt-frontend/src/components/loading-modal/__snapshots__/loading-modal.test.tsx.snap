// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingModal should match the snapshot 1`] = `
<div>
  <div
    class="sc-blHHSb dWsZIh"
    data-testid="modal-background"
  >
    <div
      class="sc-gtLWhw cgXRjV"
      data-testid="loading-indicator"
    />
    <div
      class="sc-egkSDF beDEqd"
    >
      Loading....
    </div>
  </div>
</div>
`;
